### PR TITLE
#4 Fontchain linting before building font

### PR DESCRIPTION
**Pull request by guidotheelen**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,6 +16,10 @@ especially if you are using zopflipng for compression.  Intermediate products
 (compressed image files, for example) will be put into a build subdirectory; the
 font will be at the top level.
 
+## Check artwork coverage
+
+By running `make CHECK_COVERAGE="True"` the build process is proceeded by a coverage check. The source directory is compared against the unicode standard used in [Nototools](https://github.com/googlefonts/nototools). The check flags any unexpected and missing filenames.
```
Comment (RoelN) [inline README.md:21]: preceeded

Comment (RoelN) [reply README.md:21]: unicode → Unicode

Comment (RoelN) [reply README.md:21]: "The source directory is compared" → is that the case? How does it check for emoji not in that dir?

File: check_emoji_sequences.py [hunk 2]
```diff
@@ -294,44 +294,6 @@ def _check_coverage(seq_to_filepath, unicode_version):
...
-  for seq, name in sorted(
-      unicode_data.get_emoji_zwj_sequences(age=age).iteritems()):
-    if EMOJI_VS in seq:
-      test_seq = tuple(s for s in seq if s != EMOJI_VS)
-    else:
-      test_seq = seq
-    if test_seq not in zwj_seq_without_vs:
-      print('coverage: missing (canonical) zwj sequence %s (%s)' % (
-          unicode_data.seq_to_string(seq), name))
-
```
Comment (RoelN) [inline check_emoji_sequences.py:334]: Can you explain why this chunk is now missing?

File: check_emoji_sequences.py [hunk 3]
```diff
@@ -99,7 +99,7 @@ def _check_valid_emoji_cps(sorted_seq_to_filepath, unicode_version):
   valid_cps |= TAG_SET  # used in subregion tag sequences
 
   not_emoji = {}
-  for seq, fp in sorted_seq_to_filepath.iteritems():
+  for seq, fp in sorted_seq_to_filepath.items():
```
Comment (obeleh) [inline check_emoji_sequences.py:102]: The following could be simplified by using [DefaultDict](https://docs.python.org/3.8/library/collections.html#collections.defaultdict)

File: check_emoji_sequences.py [hunk 4]
```diff
@@ -76,7 +76,7 @@ def strip_vs_map(seq_map):
 
 def _check_no_vs(sorted_seq_to_filepath):
   """Our image data does not use emoji presentation variation selectors."""
-  for seq, fp in sorted_seq_to_filepath.iteritems():
+  for seq, fp in sorted_seq_to_filepath.items():
     if EMOJI_VS in seq:
       print('check no VS: FE0F in path: %s' % fp)
```
Comment (obeleh) [inline check_emoji_sequences.py:81]: I personally dislike this style of string formatting. I'm a fan of f-strings

File: README.md (hunk 1 above)
Comment (guidotheelen) [reply README.md:21]: The Unicode files determine the emoji we need: https://unicode.org/Public/emoji/13.0/ 
If we miss an emoji in the source we get an error like this:

`coverage: missing single 1fa83 (BOOMERANG)`

File: check_emoji_sequences.py (hunk 2 above)
Comment (guidotheelen) [reply check_emoji_sequences.py:334]: This code does not exist anymore in Nototools. Now every emoji is also part of the combining sequence. So if we miss artwork we will still be able to flag it without this part.

File: check_emoji_sequences.py (hunk 4 above)
Comment (guidotheelen) [reply check_emoji_sequences.py]: Yeah I also like f-stings better! Changed them.

File: check_emoji_sequences.py (hunk 3 above)
Comment (guidotheelen) [reply check_emoji_sequences.py:102]: Good one, will fix this when we go with this approach.

**Pull request merged**